File: result/metagpt/claude/humaneval/152.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int* compare(int* game, int* guess, int size) {
    int* out = (int*)malloc(size * sizeof(int));
    for (int i = 0; i < size; i++) {
        out[i] = abs(game[i] - guess[i]);
    }
    return out;
}

File: result/metagpt/deepseek/humaneval/152.rs
pub fn compare(game: &[i32], guess: &[i32]) -> Vec<i32> {
    game.iter()
        .zip(guess.iter())
        .map(|(&g, &h)| (g - h).abs())
        .collect()
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_compare() {
        assert_eq!(compare(&[1, 2, 3], &[4, 5, 6]), vec![3, 3, 3]);
        assert_eq!(compare(&[10, 20], &[5, 25]), vec![5, 5]);
    }
}